pkg/controller/virtualmachineimage/virtualmachineimage_controller.go

```diff
@@ -126,5 +113,43 @@ func (r *ReconcileVirtualMachineImage) validateVirtualMachineImageSpec() error {
 	if !found {
 		return goerrors.New("storage request in pvc is missing")
 	}
+	src := r.getSource()
+	if src == "" {
+		return goerrors.New("no source set for vmim")
+	} else if src == sourceTypeVolume {
+		// check node name is not empty when source is hostPath volume
+		if r.vmi.Spec.Source.Volume.HostPath != nil && r.vmi.Spec.NodeName == "" {
+			return goerrors.New("when the source is hostPath volume, the node name must not be empty")
```

Comment:
If we just return error, how to the user figure out the vmim is in error state? we can see the error logs but there's no error message in vmim.

Reply:
user can check the error state through `kubectl get vmim` and the error message through `kubectl describe vmim`.
$ kubectl get vmim
```
NAME        STATE
localvmim   Error
```

$ kubectl describe vmim
```
Status:
  Conditions:
    Last Transition Time:  2020-09-18T01:16:24Z
    Message:               no source set for vmim
    Reason:                SeeMessages
    Status:                False
    Type:                  ReadyToUse
  State:                   Error

```
Isn't that enough?